File: C++/Mit Polymorphismus/Hund.cpp
#include "Hund.h"
#include <iostream>

void Hund::fressen()
{
	cout << "Knochen nagen" << endl;
}

void Hund::lautGeben()
{
	cout << "bellen" << endl;
}

void Hund::freuen()
{
	cout << "wedeln" << endl;
}

Hund::Hund(string name)
{
	this->name = name;
}

File: TypeScript/Mit Polymorphismus/Hund.ts
import {Haustier} from "./Haustier";

class Hund extends Haustier{
    constructor(n : string){
        super(n);
    }

    fressen(){
        console.log("Knochen nagen");
    }

    freuen(){
        console.log("wedeln");
    }

    lautGeben(){
        console.log("bellen");
    }
}

export {Hund};

